includes/class-freemius.php

```diff
@@ -2058,6 +2063,48 @@ function _submit_uninstall_reason_action() {
             exit;
         }
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since  2.1.4
+         */
+        function cancel_subscription_or_trial_ajax_action() {
+            $this->_logger->entrance();
+
+            $this->check_ajax_referer( 'cancel_subscription_or_trial' );
+
+            $this->cancel_subscription_or_trial( fs_request_get( 'plugin_id', $this->get_id() ) );
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since  2.1.4
+         *
+         * @param number   $plugin_id
+         * @param int|null $blog_id
+         */
+        private function cancel_subscription_or_trial( $plugin_id, $blog_id = null ) {
+            $fs = null;
+            if ( $plugin_id == $this->get_id() ) {
+                $fs = $this;
+            } else if ( $this->is_addon_activated( $plugin_id ) ) {
+                $fs = self::get_instance_by_id( $plugin_id );
+
+                $blog_id = null;
+            }
+
+            if ( ! is_null( $fs ) ) {
+                if ( $fs->is_trial() ) {

```

Comment:
@fajardoleo we only want to cancel the trial when it's a paid trial, i.e. check if `is_paid_trial()` (you may also need to adjust the logic of the dialog, we don't want to show the "cancel subscription" section if it's a trial without a payment method).